win32ss/user/ntuser/kbdlayout.c

```diff
@@ -624,8 +625,24 @@ NtUserGetKeyboardLayoutName(
 
     _SEH2_TRY
     {
-        ProbeForWrite(pwszName, KL_NAMELENGTH*sizeof(WCHAR), 1);
-        wcscpy(pwszName, pKl->spkf->awchKF);
+        ProbeForWriteUnicodeString(pustrName);
+        ProbeForWrite(pustrName->Buffer, pustrName->MaximumLength, 1);
```

Comment:
This is not a safe way to probe a `UNICODE_STRING`. `pustrName->Buffer` can change after your probe it.
You need something like
```
ProbeForWriteUnicodeString(pustrName);
ustrNameSafe = *pustrName;
ProbeForWrite(ustrNameSafe.Buffer, ustrNameSafe.MaximumLength, 1);

/* write data into use ustrNameSafe */
*pustrName = ustrNameSafe;
```

Reply:
good catch, I somehow missed it while reviewing.